prime-router/src/test/kotlin/fhirengine/engine/FhirReceiverFilterTests.kt

```diff
@@ -0,0 +1,2159 @@
+    val receiver = Receiver(
+        "myRcvr",
+        "topic",
+        Topic.TEST,
+        CustomerStatus.ACTIVE,
+        "mySchema"
+    )
+
+    private fun makeFhirEngine(metadata: Metadata, settings: SettingsProvider): FHIREngine {
+        every { reportServiceMock.getSenderName(any()) } returns "sendingOrg.sendingOrgClient"
+
+        return FHIREngine.Builder()
+            .metadata(metadata)
+            .settingsProvider(settings)
+            .databaseAccess(accessSpy)
+            .blobAccess(blobMock)
+            .azureEventService(azureEventService)
+            .reportService(reportServiceMock)
+            .build(TaskAction.route)
+    }
+
+    /**
+     * Private extension function on FHIRRouter as helper to allow setting all filter mocks in one call
+     */
+    private fun FHIRRouter.setFiltersOnEngine(
+        jurisFilter: List<String>,
+        qualFilter: List<String>,
+        routingFilter: List<String>,
+        processingModeFilter: List<String>,
+        conditionFilter: List<String> = emptyList(),
+        mappedConditionFilter: ReportStreamConditionFilter = emptyList(),
+    ) {
+        every { getJurisFilters(any(), any()) } returns jurisFilter
+        every { getQualityFilters(any(), any()) } returns qualFilter
+        every { getRoutingFilter(any(), any()) } returns routingFilter
+        every { getProcessingModeFilter(any(), any()) } returns processingModeFilter
+        every { getConditionFilter(any(), any()) } returns conditionFilter
+        every { getMappedConditionFilter(any(), any()) } returns mappedConditionFilter
+    }
+
+    @BeforeEach
+    fun reset() {
+        actionHistory.reportsIn.clear()
+        actionHistory.reportsOut.clear()
+        actionHistory.actionLogs.clear()
+        azureEventService.clear()
+        clearAllMocks()
+    }
+
+    @Test
+    fun `test applyFilters receiver setting - (reverseTheQualityFilter = true) `() {
+        val fhirData = File(QUALITY_TEST_URL).readText()
+        val bundle = FhirTranscoder.decode(fhirData)
+        val settings = FileSettings().loadOrganizations(orgWithReversedQualityFilter)
+        val jurisFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val qualFilter = listOf(PROVENANCE_COUNT_EQUAL_TO_TEN)
+        val routingFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val processingModeFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val conditionFilter = emptyList<String>()
+        val mappedConditionFilter = emptyList<ConditionFilter>()
+
+        val engine = spyk(makeFhirEngine(metadata, settings) as FHIRRouter)
+        engine.setFiltersOnEngine(
+            jurisFilter, qualFilter, routingFilter, processingModeFilter, conditionFilter, mappedConditionFilter
+        )
+
+        // act
+        val receivers = engine.findReceiversForBundle(bundle, report.id, actionHistory, Topic.FULL_ELR)
+
+        // assert
+        assertThat(receivers).isNotEmpty()
+    }
+
+    // TODO: review
```

Comment:
to-done yet?  
Edit: looks like most/all the tests here have todos on them.

Reply:
Addressed. This got stuck shelved in intellij.